internal/clusterfeature/features/vault/operator.go

```diff
@@ -0,0 +1,417 @@
+	if err != nil {
+		return errors.WrapIf(err, "failed to list auth")
+	}
+
+	if _, ok := mounts[fmt.Sprintf("%s/", path)]; ok {
+		log.Debugf("%s auth path is already in use", path)
+		return nil
+	}
+
+	return m.vaultClient.RawClient().Sys().EnableAuthWithOptions(
+		path,
+		&api.EnableAuthOptions{
+			Type: authType,
+		})
+}
+
+func getPolicyName(orgID, clusterID uint) string {
+	return fmt.Sprintf("%s_%d_%d", policyNamePrefix, orgID, clusterID)
+}
+
+func (op FeatureOperator) installOrUpdateWebhook(
+	ctx context.Context,
+	logger common.Logger,
+	orgID, clusterID uint,
+	spec vaultFeatureSpec,
+) error {
+	// create chart values
+	pipelineSystemNamespace := viper.GetString(config.PipelineSystemNamespace)
+	var chartValues = &webhookValues{
+		Env: map[string]string{
+			vaultAddressEnvKey: spec.getVaultAddress(),
+			vaultPathEnvKey:    getAuthMethodPath(orgID, clusterID),
+		},
+		NamespaceSelector: namespaceSelector{
+			MatchExpressions: []matchExpressions{
+				{
+					Key:      "name",
+					Operator: "NotIn",
+					Values: []string{
+						kubeSysNamespace,
+						pipelineSystemNamespace,
+					},
+				},
+			},
+		},
+	}
+	valuesBytes, err := json.Marshal(chartValues)
+	if err != nil {
+		logger.Debug("failed to marshal chartValues")
+		return errors.WrapIf(err, "failed to decode chartValues")
+	}
+
+	chartName, chartVersion := getChartParams()
+
+	return op.helmService.ApplyDeployment(
+		ctx,
+		clusterID,
+		pipelineSystemNamespace,
+		chartName,
+		vaultWebhookReleaseName,
+		valuesBytes,
+		chartVersion,
+	)
+}
+
+func getChartParams() (name string, version string) {
+	name = viper.GetString(config.VaultWebhookChartKey)
+	version = viper.GetString(config.VaultWebhookChartVersionKey)
+	return
+}
+
+// Deactivate deactivates the cluster feature
+func (op FeatureOperator) Deactivate(ctx context.Context, clusterID uint, spec clusterfeature.FeatureSpec) error {
+	ctx, err := op.ensureOrgIDInContext(ctx, clusterID)
+	if err != nil {
+
+		return err
+	}
+
+	if err := op.clusterService.CheckClusterReady(ctx, clusterID); err != nil {
+		return err
+	}
+
+	logger := op.logger.WithContext(ctx).WithFields(map[string]interface{}{"cluster": clusterID, "feature": featureName})
+
+	// delete deployment
+	if err := op.helmService.DeleteDeployment(ctx, clusterID, vaultWebhookReleaseName); err != nil {
+		logger.Info("failed to delete feature deployment")
+
+		return errors.WrapIf(err, "failed to uninstall feature")
+	}
+
+	logger.Info("vault webhook deployment deleted successfully")
+
+	boundSpec, err := bindFeatureSpec(spec)
+	if err != nil {
+		return clusterfeature.InvalidFeatureSpecError{
+			FeatureName: featureName,
+			Problem:     err.Error(),
+		}
+	}
+
+	if boundSpec.CustomVault.Enabled && len(boundSpec.CustomVault.TokenSecretID) != 0 {
+		cluster, err := op.clusterGetter.GetClusterByIDOnly(ctx, clusterID)
+		if err != nil {
+			return errors.New("failed to get cluster")
+		}
+
+		orgID := cluster.GetOrganizationId()
```

Comment:
If you're already calling `ensureOrgIDInContext` at the beginning, I'd use the org ID from the context instead of loading the cluster from the database again.